### ISSUE TITLE
Query node: sort forum posts search result by relevance

### ISSUE THREAD

**Issue by thesan** [Joystream/joystream#2608]
➕  we need sorting by relevance (based on full text searches) to support ticket for [ THIS ISSUE](https://github.com/Joystream/pioneer/issues/2032)

The Pioneer design now has a global search for council, proposal, and forum posts [@figma](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=4815%3A142263).

For the forum section, there should be a way to sort by relevance the result of a query to the `ForumPost` which are filtered by thread title, post text and maybe later by tag.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201709380879215/1201711172703030) by [Unito](https://www.unito.io)

**Comment by ondratra** [comment]
Filtering by the thread title, etc. should be possible already. For sorting by the relevance, we need to expose the calculated relevance value provided by PostgresSQL's `ts_rank`. For that to work properly, we need to fix issues we are having with that function as described in https://github.com/Joystream/hydra/issues/455. Also, I have an impression that we will want to include forum posts' texts in multiple fulltext searches -> that's the goal of https://github.com/Joystream/hydra/issues/457.

**Comment by thesan** [comment]
Yes AFAIK the filtering part works well, I just meant the result should be sortable by relevance. Currently it's done temporarily on the frontend.

**Comment by bedeho** [comment]
Is this still an issue?

**Comment by thesan** [comment]
> Is this still an issue?

Yes. The aggregation is still done on the front-end. Currently at most 500 results (not necessarily the most relevant) are fetch from the query node then the data is aggregated on the front-end and finally 20 results are rendered in the search side bar.

**Comment by bedeho** [comment]
Ok, adding it to `Oylmpia` scope, hopefully we can make a decision there.

**Comment by dmtrjsg** [comment]
@ondratra  assigning to you mate as per our agreement from Monday ;)
